src/ConnectedMode/Migration/IConnectedModeMigration.cs

```diff
@@ -19,10 +19,9 @@
  */
 
 using System;
-using System.ComponentModel.Composition;
 using System.Threading;
 using System.Threading.Tasks;
-using SonarLint.VisualStudio.Integration;
+using EnvDTE;
```

Comment:
The GitHub side-by-side diff looks a bit weird for this file, so you might want to ignore the "before".

I moved the "cleaner" and progress interfaces to this single file so we can see all of the related interface types together.

I moved the implementation of `ConnectedModeMigration` to a separate class. 

I think it's easier to see the bigger picture this way, but if not we can separate them out to individual files again.